pkg/controller/cloud/node_controller_test.go

```diff
@@ -331,70 +318,16 @@ func TestGCECondition(t *testing.T) {
 					},
 				},
 			},
-		},
-		Clientset:      fake.NewSimpleClientset(&v1.PodList{}),
-		DeleteWaitChan: make(chan struct{}),
-	}
-
-	factory := informers.NewSharedInformerFactory(fnh, 0)
-
-	fakeCloud := &fakecloud.Cloud{
-		InstanceTypes: map[types.NodeName]string{
-			types.NodeName("node0"): "t1.micro",
-		},
-		Addresses: []v1.NodeAddress{
-			{
-				Type:    v1.NodeHostName,
-				Address: "node0.cloud.internal",
-			},
-			{
-				Type:    v1.NodeInternalIP,
-				Address: "10.0.0.1",
-			},
-			{
-				Type:    v1.NodeExternalIP,
-				Address: "132.143.154.163",
-			},
-		},
-		Provider: "gce",
-		Err:      nil,
-	}
-
-	eventBroadcaster := record.NewBroadcaster()
-	cloudNodeController := &CloudNodeController{
-		kubeClient:   fnh,
-		nodeInformer: factory.Core().V1().Nodes(),
-		cloud:        fakeCloud,
-		recorder:     eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"}),
-	}
-	eventBroadcaster.StartLogging(klog.Infof)
-
-	cloudNodeController.AddCloudNode(context.TODO(), fnh.Existing[0])
-
-	assert.Equal(t, 1, len(fnh.UpdatedNodes), "Node was not updated")
-	assert.Equal(t, "node0", fnh.UpdatedNodes[0].Name, "Node was not updated")
-	assert.Equal(t, 2, len(fnh.UpdatedNodes[0].Status.Conditions), "No new conditions were added for GCE")
-
-	conditionAdded := false
-	for _, cond := range fnh.UpdatedNodes[0].Status.Conditions {
-		if cond.Status == "True" && cond.Type == "NetworkUnavailable" && cond.Reason == "NoRouteCreated" {
-			conditionAdded = true
-		}
-	}
-
-	assert.True(t, conditionAdded, "Network Route Condition for GCE not added by external cloud initializer")
-}
-
-// This test checks that a node with the external cloud provider taint is cloudprovider initialized and
-// and that zone labels are added correctly
-func TestZoneInitialized(t *testing.T) {
-	fnh := &testutil.FakeNodeHandler{
-		Existing: []*v1.Node{
-			{
+			updatedNode: &v1.Node{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:              "node0",
 					CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
-					Labels:            map[string]string{},
+					Labels: map[string]string{
+						"failure-domain.beta.kubernetes.io/region": "us-west",
```

Comment:
Not against a backward compatibility test but these tags are now deprecated, correct? Shouldn't we also have a version of this test using the GA labels?

Reply:
Actually is topology.kubernetes.io/region the GA version? Some of our release notes reference "failure-domain.kubernetes.io/region", which matched my memory but the code seems to indicate "topology.kubernetes.io/region".

Reply:
It would still be good to have a test case which does not use "failure-domain.beta.kubernetes.io".

Reply:
We currently add both labels for compatibility and won't be removing the beta ones  until v1.21. This test checks for both.

Reply:
Correct me if I'm wrong, but I thought it was the cloud provider who would be adding these labels. As such it would be good to make sure that we do the right thing when we both styles of label are provided and also when only the new labels are provided.

Reply:
> Correct me if I'm wrong, but I thought it was the cloud provider who would be adding these labels.

The cloud provider provides the zone/region values, but the upstream controller (cloud node controller) will always set both beta/GA labels. There's no way at the moment to only add the GA ones.